boards/arm/sparkfun_micromod_stm32/sparkfun_micromod_stm32.dts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2022 Alvaro Viebrantz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/f4/stm32f405Xg.dtsi>
+#include <st/f4/stm32f405rgtx-pinctrl.dtsi>
+
+/ {
+	model = "Sparkfun MicroMod STM32 Processor";
+	compatible = "st,sparkfun_micromod_stm32", "st,stm32f405";
+
+	chosen {
+		zephyr,console = &usart2;
+		zephyr,shell-uart = &usart2;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,ccm = &ccm0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led: led {
+			gpios = <&gpioa 15 GPIO_ACTIVE_HIGH>;
+			label = "User LED";
+		};
+	};
+
+	aliases {
+		led0 = &led;
+	};
+};
+
+&clk_hse {
+	clock-frequency = <DT_FREQ_M(12)>;
+	status = "okay";
+};
+
+&pll {
+	div-m = <12>;
+	mul-n = <336>;
+	div-p = <2>;
+	div-q = <7>;
+	clocks = <&clk_hse>;
+	status = "okay";
+};
+
+&rcc {
+	clocks = <&pll>;
+	clock-frequency = <DT_FREQ_M(168)>;
+	ahb-prescaler = <1>;
+	apb1-prescaler = <4>;
+	apb2-prescaler = <2>;
+};
+
+&usart2 {
+	pinctrl-0 = <&usart2_tx_pa2 &usart2_rx_pa3>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+	status = "okay";
+};
+
+&i2c1 {
+	pinctrl-0 = <&i2c1_scl_pb6 &i2c1_sda_pb7>;
+	pinctrl-names = "default";
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+&i2c2 {
+	pinctrl-0 = <&i2c2_scl_pb10 &i2c2_sda_pb11>;
+	pinctrl-names = "default";
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+&spi1 {
+	pinctrl-0 = <&spi1_sck_pa5 &spi1_miso_pa6 &spi1_mosi_pa7>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&rtc {
+	status = "okay";
+};
```

Comment:
I'm not seeing any mention to RTC otherwise in the board porting. Is that intended to enable it?

Reply:
You can test it using counter api test /sample

Reply:
Tested here the `samples/drivers/counter/alarm` and it worked.

`west build -b sparkfun_micromod_stm32 samples/drivers/counter/alarm -p`

```
*** Booting Zephyr OS build zephyr-v3.0.0-3807-g6ec92aa5232a  ***
Counter alarm sample

Set alarm in 2 sec (2 ticks)
!!! Alarm !!!
Now: 3
Set alarm in 4 sec (4 ticks)
!!! Alarm !!!
Now: 8
Set alarm in 8 sec (8 ticks)
```